src/main/java/io/reactivex/schedulers/Schedulers.java

```diff
@@ -43,56 +45,105 @@
         NEW_THREAD = RxJavaPlugins.initNewThreadScheduler(NewThreadScheduler.instance());
     }
     
-    /*
-     * TODO I started to like enums for singletons and non-instantiatable
-     * utility classes, but since this is part of the public API,
-     * that would act quite unorthodoxically.
-     */
     private Schedulers() {
         throw new IllegalStateException("No instances");
     }
     
+    /**
+     * Creates and returns a {@link Scheduler} intended for computational work.
+     * <p>
+     * This can be used for event-loops, processing callbacks and other computational work.
+     * <p>
+     * Do not perform IO-bound work on this scheduler. Use {@link #io()} instead.
+     * <p>
+     * Unhandled errors will be delivered to the scheduler Thread's {@link java.lang.Thread.UncaughtExceptionHandler}.
+     *
+     * @return a {@link Scheduler} meant for computation-bound work
+     */
     public static Scheduler computation() {
         return RxJavaPlugins.onComputationScheduler(COMPUTATION);
     }
     
+    /**
+     * Creates and returns a {@link Scheduler} intended for IO-bound work.
+     * <p>
+     * The implementation is backed by an {@link Executor} thread-pool that will grow as needed.
+     * <p>
+     * This can be used for asynchronously performing blocking IO.
+     * <p>
+     * Do not perform computational work on this scheduler. Use {@link #computation()} instead.
+     * <p>
+     * Unhandled errors will be delivered to the scheduler Thread's {@link java.lang.Thread.UncaughtExceptionHandler}.
+     *
+     * @return a {@link Scheduler} meant for IO-bound work
+     */
     public static Scheduler io() {
         return RxJavaPlugins.onIoScheduler(IO);
     }
     
-    public static TestScheduler test() {
+    /**
+     * Creates and returns a {@code TestScheduler}, which is useful for debugging. It allows you to test
+     * schedules of events by manually advancing the clock at whatever pace you choose.
+     *
+     * @return a {@code TestScheduler} meant for debugging
+     */
+    public static TestScheduler test() { // NOPMD
         return new TestScheduler();
```

Comment:
Always!

On Fri, Aug 19, 2016 at 4:41 PM David Karnok <notifications@github.com>
wrote:

> In src/main/java/io/reactivex/schedulers/Schedulers.java
> <https://github.com/ReactiveX/RxJava/pull/4378#discussion_r75548361>:
>
> >      public static Scheduler io() {
> >          return RxJavaPlugins.onIoScheduler(IO);
> >      }
> >
> > -    public static TestScheduler test() {
> > +    /**
> > +     * Creates and returns a {@code TestScheduler}, which is useful for debugging. It allows you to test
> > +     * schedules of events by manually advancing the clock at whatever pace you choose.
> > +     *
> > +     * @return a {@code TestScheduler} meant for debugging
> > +     */
> > +    public static TestScheduler test() { // NOPMD
> >          return new TestScheduler();
>
> I see. Will you post a PR?
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/ReactiveX/RxJava/pull/4378/files/dcdf77a7e2fd6bc836ba63e443424b405321ebcc#r75548361>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAEEEe2CE0q5ZDuQa2XTRs0gHCYT_lLNks5qhhUUgaJpZM4JozrR>
> .
>